src/pyggdrasil/tree_inference/_tree.py

```diff
@@ -310,3 +321,31 @@ def is_same_tree(tree1: Tree, tree2: Tree) -> bool:
         ) and jnp.all(tree1.labels == tree2_reordered.labels)
 
     return bool(result)
+
+
+def is_valid_tree(tree: Tree) -> bool:
```

Comment:
Good idea, it should catch some bugs. Do we know that `to_TreeNode` explicitly check for these four things or we hope it does it implicitly?

Reply:
Yes, converting to `TreeNode` from `Tree` is what so far has caught all the broken trees. To convert it needs to check for the 1) root, 2) self-loops would lead to recursion failing 3)  not connected components leads to multiple roots, also caught